metrics-core/src/main/java/software/amazon/swage/metrics/record/cloudwatch/MetricDataAggregator.java

```diff
@@ -83,15 +84,11 @@ public void add(
         //TODO: avoid doing this every time for a context - caching, or?
         List<Dimension> dimensions = dimensionMapper.getDimensions(name, context);
 
-        DatumKey key = new DatumKey(name.toString(), unit, dimensions);
+        DatumKey key = new DatumKey(name.toString(), unit, dimensions, value);
         statisticsMap.merge(
                 key,
-                new StatisticSet()
-                        .withMaximum(value)
-                        .withMinimum(value)
-                        .withSampleCount(1D)
-                        .withSum(value),
-                MetricDataAggregator::sum);
+                1L,
+                (a, b) -> a + b);
```

Comment:
Should we handle overflows?

Reply:
No. The value of the map is the count of occurrences of a specific value, which would take an unbelievable amount of time to overflow an Integer, let alone a Long. This is especially true when this *(name,unit,dimensions,value)->count* map is only used for `record()` and not `count()`.

Reply:
Though when I add a map for *(name,unit,dimensions)->sum* (used for `count()` and not `record()`), the question arises again. However, I can't imagine `count()` should ever be called with a value beside 0 or 1. And in that case, Integer should be sufficient. I'll stick with Long, though, because that's what conventionally is used for this purpose (Guava's AtomicLongMap, Java 1.8's Collectors::counting).